examples/dnn_semantic_segmentation_train_ex.cpp

```diff
@@ -0,0 +1,317 @@
+    extract_image_chip(label_image, chip_details, crop.second, interpolate_nearest_neighbor());
+
+    // Also randomly flip the input image and the labels.
+    if (rnd.get_random_double() > 0.5) {
+        crop.first = fliplr(crop.first);
+        crop.second = fliplr(crop.second);
+    }
+
+    // And then randomly adjust the colors.
+    apply_random_color_offset(crop.first, rnd);
+}
+
+// ----------------------------------------------------------------------------------------
+
+struct image_info
+{
+    string image_filename;
+    string label_filename;
+};
+
+std::vector<image_info> get_pascal_voc2012_listing(
+    const std::string& voc2012_folder,
+    const std::string& file = "train" // "train", "trainval", or "val"
+)
+{
+    std::ifstream in(voc2012_folder + "/ImageSets/Segmentation/" + file + ".txt");
+
+    std::vector<image_info> results;
+
+    while (in) {
+        std::string basename;
+        in >> basename;
+
+        if (!basename.empty()) {
+            image_info image_info;
+            image_info.image_filename = voc2012_folder + "/JPEGImages/" + basename + ".jpg";
+            image_info.label_filename = voc2012_folder + "/SegmentationClass/" + basename + ".png";
+            results.push_back(image_info);
+        }
+    }
+
+    return results;
+}
+
+std::vector<image_info> get_pascal_voc2012_train_listing(
+    const std::string& voc2012_folder
+)
+{
+    return get_pascal_voc2012_listing(voc2012_folder, "train");
+}
+
+std::vector<image_info> get_pascal_voc2012_val_listing(
+    const std::string& voc2012_folder
+)
+{
+    return get_pascal_voc2012_listing(voc2012_folder, "val");
+}
+
+// ----------------------------------------------------------------------------------------
+
+const Voc2012class& find_voc2012_class(const dlib::rgb_pixel& rgb_label)
+{
+    return find_voc2012_class(
+        [&rgb_label](const Voc2012class& voc2012class) {
+            return rgb_label == voc2012class.rgb_label;
+        }
+    );
+}
+
+inline uint16_t rgb_label_to_index_label(const dlib::rgb_pixel& rgb_label)
+{
+    return find_voc2012_class(rgb_label).index;
+}
+
+void rgb_label_image_to_index_label_image(const dlib::matrix<dlib::rgb_pixel>& rgb_label_image, dlib::matrix<uint16_t>& index_label_image)
+{
+    const long nr = rgb_label_image.nr();
+    const long nc = rgb_label_image.nc();
+
+    index_label_image.set_size(nr, nc);
+
+    for (long r = 0; r < nr; ++r) {
+        for (long c = 0; c < nc; ++c) {
+            index_label_image(r, c) = rgb_label_to_index_label(rgb_label_image(r, c));
+        }
+    }
+}
+
+// ----------------------------------------------------------------------------------------
+
+double calculate_accuracy(anet_type& anet, const std::vector<image_info>& dataset)
+{
+    int num_right = 0;
+    int num_wrong = 0;
+
+    matrix<rgb_pixel> input_image;
+    matrix<rgb_pixel> rgb_label_image;
+    matrix<uint16_t> index_label_image;
+
+    for (const auto& image_info : dataset) {
```

Comment:
Oh and put { on the line by itself so it's consistent with the rest of dlib.

Reply:
Fixed.

I wonder if it'd be straightforward to add a linter on either Travis or AppVeyor to check this stuff, so that you wouldn't need to keep mentioning about it.